test/test_modules.py

```diff
@@ -95,22 +126,63 @@ def test_factory_kwargs(self, device, dtype, module_info):
+
+    @onlyCUDA
+    @modules(module_db)
+    def test_multiple_device_transfer(self, device, dtype, module_info):
+        module_cls = module_info.module_cls
+        module_inputs_device = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                              requires_grad=False)
+        module_inputs_cpu = module_info.module_inputs_func(module_info, device="cpu", dtype=dtype,
+                                                           requires_grad=False)
+        for module_input_device, module_input_cpu in zip(module_inputs_device, module_inputs_cpu):
+            if module_input_device.forward_input is None:
+                continue
+
+            with freeze_rng_state():
+                # === Instantiate the module. ===
+                args, kwargs = module_input_device.constructor_input.args, module_input_device.constructor_input.kwargs
+                m = module_cls(*args, **kwargs)
+                m.to(device).to(dtype)
+
+                # === Do forward pass on GPU ===
+                input_device_args = module_input_device.forward_input.args
+                input_device_kwargs = module_input_device.forward_input.kwargs
+                m(*input_device_args, **input_device_kwargs)
+                self._assert_module_parameters_and_buffer_are(m, device, dtype, 0)
+
+                # === Move to CPU ===
+                input_cpu_args = module_input_cpu.forward_input.args
+                input_cpu_kwargs = module_input_cpu.forward_input.kwargs
+                m.cpu()
+                m(*input_cpu_args, **input_cpu_kwargs)
+                self._assert_module_parameters_and_buffer_are(m, "cpu", dtype)
+
+                # === Move back to GPU and forward pass ===
+                m.cuda()
+                m(*input_device_args, **input_device_kwargs)
+                self._assert_module_parameters_and_buffer_are(m, device, dtype, 0)
+
+                if torch.cuda.device_count() >= 2:
+                    # === test cross-GPU transfer works
+                    def _to_device(objs):
+                        if isinstance(objs, (tuple, list)):
+                            return type(objs)(_to_device(item) for item in objs)
+                        elif isinstance(objs, dict):
+                            return {name: _to_device(item) for name, item in objs.items()}
+                        elif isinstance(objs, torch.Tensor):
+                            return objs.cuda(1)
+                        else:
+                            return objs
+                    input_device_1_args = _to_device(input_device_args)
+                    input_device_1_kwargs = _to_device(input_device_kwargs)
+
+                    m.cuda(1)
+                    with torch.cuda.device(1):
+                        m(*input_device_1_args, **input_device_1_kwargs)
+                    self._assert_module_parameters_and_buffer_are(m, device, dtype, 1)
```

Comment:
@thomasjpfan FYI device here is `'cuda:0'` which is causing failing checks. I don't see why a separate device ID check is necessary since the device name includes IDs.